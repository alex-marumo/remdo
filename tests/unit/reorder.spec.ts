import "./common";
import { it } from "vitest";

it("reorder flat", async ({ load, editor, expect, lexicalUpdate }) => {
  const { note0 } = load("flat");
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => note0.moveDown());
  await expect(editor).toMatchFileSnapshot("note0-down-x1.yml");

  lexicalUpdate(() => note0.moveDown());
  await expect(editor).toMatchFileSnapshot("note0-down-x2.yml");

  lexicalUpdate(() => note0.moveDown()); //noop
  await expect(editor).toMatchFileSnapshot("note0-down-x2.yml");

  lexicalUpdate(() => note0.moveUp());
  await expect(editor).toMatchFileSnapshot("note0-down-x1.yml");

  lexicalUpdate(() => note0.moveUp());
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => note0.moveUp()); //noop
  await expect(editor).toMatchFileSnapshot("base.yml");
});

it("reorder tree", async ({ load, editor, expect, lexicalUpdate }) => {
  const { note0, subNote0 } = load("tree");
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => note0.moveDown());
  await expect(editor).toMatchFileSnapshot("note0-down.yml");

  lexicalUpdate(() => note0.moveDown()); //noop
  await expect(editor).toMatchFileSnapshot("note0-down.yml");

  lexicalUpdate(() => note0.moveUp());
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => note0.moveUp()); //noop
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => subNote0.moveUp()); //noop
  await expect(editor).toMatchFileSnapshot("base.yml");

  lexicalUpdate(() => subNote0.moveDown()); //noop
  await expect(editor).toMatchFileSnapshot("base.yml");
});
