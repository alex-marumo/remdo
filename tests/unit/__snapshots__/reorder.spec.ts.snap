// Vitest Snapshot v1

exports[`reorder flat > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - text: note2
        type: text
"
`;

exports[`reorder flat > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - text: note2
        type: text
"
`;

exports[`reorder flat > base 3`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - text: note2
        type: text
"
`;

exports[`reorder flat > note0 down 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - text: note0
        type: text
    - type: listitem
      value: 3
      children:
      - text: note2
        type: text
"
`;

exports[`reorder flat > note0 down 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - text: note0
        type: text
    - type: listitem
      value: 3
      children:
      - text: note2
        type: text
"
`;

exports[`reorder flat > note0 down x2 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - text: note2
        type: text
    - type: listitem
      value: 3
      children:
      - text: note0
        type: text
"
`;

exports[`reorder flat > note0 down x2 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - text: note2
        type: text
    - type: listitem
      value: 3
      children:
      - text: note0
        type: text
"
`;

exports[`reorder tree > base 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
"
`;

exports[`reorder tree > base 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
"
`;

exports[`reorder tree > base 3`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
"
`;

exports[`reorder tree > base 4`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
"
`;

exports[`reorder tree > base 5`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note0
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
    - type: listitem
      value: 2
      children:
      - text: note1
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
"
`;

exports[`reorder tree > note0 down 1`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
    - type: listitem
      value: 2
      children:
      - text: note0
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
"
`;

exports[`reorder tree > note0 down 2`] = `
"root:
  type: root
  children:
  - type: list
    children:
    - type: listitem
      value: 1
      children:
      - text: note1
        type: text
    - type: listitem
      value: 2
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 1
            type: text
    - type: listitem
      value: 2
      children:
      - text: note0
        type: text
    - type: listitem
      value: 3
      children:
      - type: list
        children:
        - indent: 1
          type: listitem
          value: 1
          children:
          - text: sub note 0
            type: text
"
`;
